### PR TITLE
updated kotlin version. Updated configuration for maven publishing

### PR DESCRIPTION
**Pull request by EvgeniyKurinnoy**

### CODE REVIEW

File: common-build-scripts/src/main/kotlin/PomConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    licenses {
+        license {
+            name.set("The Apache License, Version 2.0")
+            url.set("http://www.apache.org/licenses/LICENSE-2.0.txt")
+        }
+    }
+
+    developers {
+        listOf(
+            "Evgeniy Kurinnoy" to "evgeniy.kurinnoy@welltech.com",
```
Comment (a-trykashnyi-welltech) [inline common-build-scripts/src/main/kotlin/PomConfiguration.kt]: Why do we need add these emails?

File: recomposition-logger-plugin/compiler-plugin/build.gradle.kts [hunk 2]
```diff
@@ -1,52 +1,69 @@
...
 tasks.register("buildAndPublishToMavenLocal") {
-  dependsOn(tasks.named("assemble"), tasks.named("publishToMavenLocal"))
+    dependsOn(tasks.named("assemble"), tasks.named("publishDefaultPublicationToMavenLocal"))
+}
+
+tasks.register("buildAndPublishToMavenRepository") {
+    dependsOn(tasks.named("assemble"), tasks.named("publishDefaultPublicationToMavenRepository"))
+}
+
+tasks.register("buildAndPublishToMaven2Repository") {
```
Comment (a-trykashnyi-welltech) [inline recomposition-logger-plugin/compiler-plugin/build.gradle.kts]: Mave2?

File: common-build-scripts/src/main/kotlin/PomConfiguration.kt (hunk 1 above)
Comment (EvgeniyKurinnoy) [reply common-build-scripts/src/main/kotlin/PomConfiguration.kt]: Removed

File: recomposition-logger-plugin/compiler-plugin/build.gradle.kts (hunk 2 above)
Comment (EvgeniyKurinnoy) [reply recomposition-logger-plugin/compiler-plugin/build.gradle.kts]: Renamed to `buildAndPublishToSnapshotRepository`

**Review by a-trykashnyi-welltech — APPROVED** [review]

**Pull request merged**